src/Prelude.hs

```diff
@@ -16,3 +17,6 @@ endLine = "\n"
 
 memptyIfFalse :: Monoid m => Bool -> m -> m
 memptyIfFalse p val = if p then val else mempty
+
+mapAndList :: (Foldable t, Functor t) => (a -> b) -> t a -> [b]
+mapAndList f = toList . fmap f
```

Comment:
I feel like writing this function in the following way is better:

```haskell
toListMap :: Foldable f => (a -> b) -> f a -> [b]
toListMap f = map f . toList
```

In this case you're able to use `map` function specialised to the list data type which potentially can be more performant (doesn't matter in our case, but still). Also, it will make constraints less strict.